src/Microsoft.ServiceFabric.Services.Queryable/Controller/QueryableController.cs

```diff
@@ -95,12 +198,41 @@ private IHttpActionResult HandleException(Exception e, Uri serviceUri)
 			}
 		}
 
+		private static async Task<IEnumerable<T>> GetServiceProxyForPartitionAsync<T>(Uri serviceUri, Guid partitionId)
+			where T : IService
+		{
+			using (var client = new FabricClient())
+			{
+				var partitions = await client.QueryManager.GetPartitionListAsync(serviceUri).ConfigureAwait(false);
+				var matchingPartitions =
+					partitions.Where(p => p.PartitionInformation.Id == partitionId || partitionId == Guid.Empty);
+				return matchingPartitions.Select(p => CreateServiceProxy<T>(serviceUri, p));
+			}
+		}
+
+		private static async Task<T> GetServiceProxyForAddAsync<T>(Uri serviceUri, Guid partitionId) where T : IService
+		{
+			using (var client = new FabricClient())
+			{
+				var partitions = await client.QueryManager.GetPartitionListAsync(serviceUri).ConfigureAwait(false);
+				var matchingPartitions = partitions.Where(p => p.PartitionInformation.Id == partitionId);
+				if (partitionId == Guid.Empty)
+				{
+					return CreateServiceProxy<T>(serviceUri, partitions.First());
```

Comment:
This doesn't seem like the optimal spread of partitions. Can you pick one at random? Do you need the user of the library to define a random partition method for you to call?